Client/Utils/FaviconFetcher.swift

```diff
@@ -173,37 +173,34 @@ open class FaviconFetcher: NSObject, XMLParserDelegate {
     func getFavicon(_ siteUrl: URL, icon: Favicon, profile: Profile) -> Deferred<Maybe<Favicon>> {
         let deferred = Deferred<Maybe<Favicon>>()
         let url = icon.url
-        let manager = SDWebImageManager.shared()
         let site = Site(url: siteUrl.absoluteString, title: "")
 
-        var fav = Favicon(url: url)
+        var favicon = Favicon(url: url)
         if let url = url.asURL {
-            var fetch: SDWebImageOperation?
-            fetch = manager.loadImage(with: url,
-                options: .lowPriority,
-                progress: { (receivedSize, expectedSize, _) in
-                    if receivedSize > FaviconHandler.MaximumFaviconSize || expectedSize > FaviconHandler.MaximumFaviconSize {
-                        fetch?.cancel()
-                    }
-                },
-                completed: { (img, _, _, _, _, url) in
-                    guard let url = url else {
-                        deferred.fill(Maybe(failure: FaviconError()))
-                        return
-                    }
-                    fav = Favicon(url: url.absoluteString)
-
-                    if let img = img {
-                        fav.width = Int(img.size.width)
-                        fav.height = Int(img.size.height)
-                        FaviconMO.add(fav, forSiteUrl: siteUrl)
-                    } else {
-                        fav.width = 0
-                        fav.height = 0
-                    }
+            var imageOperation: SDWebImageOperation?
 
-                    deferred.fill(Maybe(success: fav))
-            })
+            let onProgress: ImageCacheProgress = { (receivedSize, expectedSize, _) -> Void in
```

Comment:
`(receivedSize, expectedSize, _) -> Void in` -> `receivedSize, expectedSize, _in`